tiledb/sm/rest/curl.cc

```diff
@@ -241,6 +258,27 @@ Status Curl::init(
   return Status::Ok();
 }
 
+size_t Curl::OnReceiveData(
+    void* res_data, size_t tSize, size_t tCount, void* userdata) {
+  size_t length = tSize * tCount, index = 0;
+  while (index < length) {
+    unsigned char* temp = (unsigned char*)res_data + index;
```

Comment:
instead of casting `res_data` to an `unsigned_char*` in multiple places, let's just create a `unsigned char* buffer = static_cast<unsigned char*>(res_data)` and use that instead.